### PR TITLE
add button icon

### PR DESCRIPTION
**Pull request by z695245771**

### CODE REVIEW

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -125,9 +126,13 @@ export default function Button({
     });
   }
 
+  const NAV_ICON_SIZE = '16px';
+
   return (
     <CustomTag {...props}>
-      {icon && <img src={icon} alt='button icon' />}
+      {icon?.slice(0, 4) === 'http'
```
Comment (wsh941102) [inline src/components/Button/index.tsx:133]: sometime img address is not begin with 'http'

File: src/components/Button/helper.tsx [hunk 2]
```diff
@@ -51,3 +52,13 @@ export const colorType = (colorHex: string): string => {
...
 };
+
+export const iconType = (icon: string): boolean => {
+  if (
+    Iconslist.basic.includes(icon) ||
+    Iconslist.outline.includes(icon) ||
+    Iconslist.filled.includes(icon)
+  )
+    return true;
+  return false;
```
Comment (mxdi9i7) [inline src/components/Button/helper.tsx:63]: you only need 1 return

File: src/components/Button/index.tsx [hunk 3]
```diff
@@ -125,9 +127,13 @@ export default function Button({
...
   }
 
+  const NAV_ICON_SIZE = '16px';
+
   return (
     <CustomTag {...props}>
-      {icon && <img src={icon} alt='button icon' />}
+      {iconType(`${icon}`)
+        ? icon && <Icon name={icon} size={NAV_ICON_SIZE} />
+        : icon && <img src={icon} alt='button icon' />}
```
Comment (mxdi9i7) [inline src/components/Button/index.tsx:136]: The image icon should not be a fallback

File: src/components/Button/helper.tsx [hunk 4]
```diff
@@ -51,3 +52,13 @@ export const colorType = (colorHex: string): string => {
   const result = '#' + colorHex;
   return result;
 };
+
+export const iconType = (icon: string): boolean => {
```
Comment (mxdi9i7) [inline src/components/Button/helper.tsx:56]: We don't technically need this, we can have this in a Type or Interface

**Pull request merged**